### PR TITLE
Add HCP Packer registry image metadata to builder artifacts

### PR DESCRIPTION
**Pull request by nywilken**
This change adds support for publishing HCP Packer registry images from existing builder artifacts with the Provider name set to "gce".  The images currently don't have any label information but would

### CODE REVIEW

File: builder/googlecompute/artifact.go [hunk 1]
```diff
@@ -43,8 +45,14 @@ func (a *Artifact) String() string {
 }
 
 func (a *Artifact) State(name string) interface{} {
-	if _, ok := a.StateData[name]; ok {
-		return a.StateData[name]
+	if name == registryimage.ArtifactStateURI {
```
Comment (nywilken) [inline builder/googlecompute/artifact.go:48]: There's some rich metadata that we can extract from the Artifact config here if set. I will revisit this tomorrow.

Comment (SwampDragons) [reply builder/googlecompute/artifact.go:48]: Sounds good.

**Review by SwampDragons — APPROVED** [review]
Works as advertised. Used this plugin to run a gce build, upload it to PAR, and then was able to pull from PAR to use output from original build as a Packer source.

**Review by azr — APPROVED** [review]
LGTM ! 🙂

File: builder/googlecompute/artifact.go (hunk 1 above)
Comment (nywilken) [reply builder/googlecompute/artifact.go:50]: Labels added. Thanks again for the review.

**Pull request merged**